Showstopper:  9.4 after upgrading I am no longer able to edit module settings

<!-- 
  Please read contribution guideline first: https://github.com/dnnsoftware/Dnn.Platform/blob/development/CONTRIBUTING.md 
  Any potential security issues should be sent to security@dnnsoftware.com, rather than posted on GitHub
-->

## Description of bug
Something is very off with the edit module settings.  On two instances of DNN one that is pretty bare and one with a load of modules... both same issue once you click on Edit Settings one a module on a page the popup window for editing the settings of a module such as permissions does not work.

Default.aspx?tabid=96&error=Sequence+contains+more+than+one+matching+element

## Steps to reproduce
List the steps to reproduce the behavior:
1. Upgrade or possibly just install DNN 9.4
2. Click on Edit Settings on any module on a page with edit mode
3. Bugs out the website
4. Test on many pages popup never opens... See error

## Current result
Edit module settings window popup errors out.
Default.aspx?tabid=39&error=Sequence+contains+more+than+one+matching+element

Message:Sequence contains more than one matching element

StackTrace:

InnerMessage:Sequence contains more than one matching element
AbsoluteURL:/Default.aspx
DefaultDataProvider:DotNetNuke.Data.SqlDataProvider, DotNetNuke
ExceptionGUID:88e591f2-0a59-4ab2-a3d2-954cd48d31c9
AssemblyVersion:9.4.0
PortalId:0
UserId:1
TabId:39
RawUrl:/Forum/ctl/Module/ModuleId/409?ReturnURL=/Forum&popUp=true
InnerStackTrace:

at System.Linq.Enumerable.SingleOrDefault[TSource](IEnumerable`1 source, Func`2 predicate) at DotNetNuke.Web.UI.WebControls.Internal.DnnComboBox.RegisterRequestResources() at DotNetNuke.Web.UI.WebControls.Internal.DnnComboBox.OnPreRender(EventArgs e) at System.Web.UI.Control.PreRenderRecursiveInternal() at System.Web.UI.Control.PreRenderRecursiveInternal() at System.Web.UI.Control.PreRenderRecursiveInternal() at System.Web.UI.Control.PreRenderRecursiveInternal() at System.Web.UI.Control.PreRenderRecursiveInternal() at System.Web.UI.Control.PreRenderRecursiveInternal() at System.Web.UI.Control.PreRenderRecursiveInternal() at System.Web.UI.Control.PreRenderRecursiveInternal() at System.Web.UI.Control.PreRenderRecursiveInternal() at System.Web.UI.Control.PreRenderRecursiveInternal() at System.Web.UI.Control.PreRenderRecursiveInternal() at System.Web.UI.Control.PreRenderRecursiveInternal() at System.Web.UI.Control.PreRenderRecursiveInternal() at System.Web.UI.Control.PreRenderRecursiveInternal() at System.Web.UI.Page.ProcessRequestMain(Boolean includeStagesBeforeAsyncPoint, Boolean includeStagesAfterAsyncPoint)




* [ ] 10.0.0 nightly build
* [ ] 9.4.1 nightly build
* [x ] 9.4.0 latest supported release

## Affected browser
<!-- 
  Check all that apply and add more if necessary.
  If possible, please also specify exact versions and mention the operating system
-->

* [ ] Chrome
* [ x] Firefox
* [ ] Safari
* [ ] Internet Explorer
* [ ] Edge

I get the same error after upgrading from 9.3.2 to 9.4.0. After clicking the settings gear I get a red banner saying "A critical error has occurred. Please check the Event Viewer for further details." The Event Viewer shows the same error as thabaum showed above. 
Portals\_default\logs shows several errors, such as:
`System.IO.FileNotFoundException: Could not load file or assembly 'Dnn.PersonaBar.Pages' or one of its dependencies. The system cannot find the file specified.
File name: 'Dnn.PersonaBar.Pages'`
`[ERROR] Dnn.PersonaBar.Library.Controllers.PersonaBarController - System.ArgumentNullException: Value cannot be null.`
`[ERROR] DotNetNuke.Framework.Reflection - Dnn.PersonaBar.Licensing.MenuControllers.HostMenuController, Dnn.PersonaBar.Licensing
System.IO.FileNotFoundException: Could not load file or assembly 'Dnn.PersonaBar.Licensing' or one of its dependencies. The system cannot find the file specified.`
I'll attach the log file.

[2019.09.15.log.resources.txt](https://github.com/dnnsoftware/Dnn.Platform/files/3613970/2019.09.15.log.resources.txt)

--- a/DNN Platform/Library/Framework/JavaScriptLibraries/IJavaScriptLibraryController.cs
+++ b/DNN Platform/Library/Framework/JavaScriptLibraries/IJavaScriptLibraryController.cs
@@ -26,14 +26,32 @@ namespace DotNetNuke.Framework.JavaScriptLibraries
 {
     public interface IJavaScriptLibraryController
     {
+        /// <summary>Delete the library reference from the database</summary>
+        /// <param name="library">Library to be deleted</param>
         void DeleteLibrary(JavaScriptLibrary library);
 
+        /// <summary>Get information about the latest version of a <see cref="JavaScriptLibrary"/> that matches the given <paramref name="predicate"/></summary>
+        /// <param name="predicate">A function used to filter the library</param>
+        /// <example>
+        /// JavaScriptLibraryController.Instance.GetLibrary(l => string.Equals(l.LibraryName, "Knockout", StringComparison.OrdinalIgnoreCase))
+        /// </example>
+        /// <returns>The highest version <see cref="JavaScriptLibrary"/> instance that matches the <paramref name="predicate"/>, or <c>null</c> if no library matches</returns>
         JavaScriptLibrary GetLibrary(Func<JavaScriptLibrary, bool> predicate);
 
+        /// <summary>Gets all of the <see cref="JavaScriptLibrary"/> instances matching the given <paramref name="predicate"/></summary>
+        /// <param name="predicate">A function used to filter the library</param>
+        /// <example>
+        /// JavaScriptLibraryController.Instance.GetLibraries(l => string.Equals(l.LibraryName, "Knockout", StringComparison.OrdinalIgnoreCase))
+        /// </example>
+        /// <returns>A sequence of <see cref="JavaScriptLibrary"/> instances</returns>
         IEnumerable<JavaScriptLibrary> GetLibraries(Func<JavaScriptLibrary, bool> predicate);
 
+        /// <summary>Gets all of the <see cref="JavaScriptLibrary"/> instances</summary>
+        /// <returns>A sequence of <see cref="JavaScriptLibrary"/> instances</returns>
         IEnumerable<JavaScriptLibrary> GetLibraries();
 
+        /// <summary>Save a library to the database</summary>
+        /// <param name="library">Library to be saved</param>
         void SaveLibrary(JavaScriptLibrary library);
     }
 }

--- a/DNN Platform/Library/Framework/JavaScriptLibraries/JavaScript.cs
+++ b/DNN Platform/Library/Framework/JavaScriptLibraries/JavaScript.cs
@@ -87,7 +87,7 @@ namespace DotNetNuke.Framework.JavaScriptLibraries
         /// <returns></returns>
         public static string Version(String jsname)
         {
-            JavaScriptLibrary library = JavaScriptLibraryController.Instance.GetLibrary(l => l.LibraryName.Equals(jsname, StringComparison.OrdinalIgnoreCase));
+            JavaScriptLibrary library = GetHighestVersionLibrary(jsname);
             return library != null ? Convert.ToString(library.Version) : String.Empty;
         }
 

--- a/DNN Platform/Library/Framework/JavaScriptLibraries/JavaScriptLibraryController.cs
+++ b/DNN Platform/Library/Framework/JavaScriptLibraries/JavaScriptLibraryController.cs
@@ -61,7 +61,7 @@ namespace DotNetNuke.Framework.JavaScriptLibraries
         /// <returns>instance of JavaScriptLibrary</returns>
         public JavaScriptLibrary GetLibrary(Func<JavaScriptLibrary, bool> predicate)
         {
-            return GetLibraries().SingleOrDefault(predicate);
+            return GetLibraries(predicate).OrderByDescending(l => l.Version).FirstOrDefault();
         }
 
         /// <summary>

--- a/DNN Platform/Library/Framework/JavaScriptLibraries/JavaScriptLibraryController.cs
+++ b/DNN Platform/Library/Framework/JavaScriptLibraries/JavaScriptLibraryController.cs
@@ -44,40 +44,38 @@ namespace DotNetNuke.Framework.JavaScriptLibraries
 
         #region IJavaScriptController Implementation
 
-        /// <summary>
-        /// delete the library reference from the database
-        /// </summary>
-        /// <param name="library">library to be deleted</param>
+        /// <summary>Delete the library reference from the database</summary>
+        /// <param name="library">Library to be deleted</param>
         public void DeleteLibrary(JavaScriptLibrary library)
         {
             DataProvider.Instance().ExecuteNonQuery("DeleteJavaScriptLibrary", library.JavaScriptLibraryID);
             ClearCache();
         }
 
-        /// <summary>
-        /// return a library reference
-        /// </summary>
-        /// <param name="predicate">predicate to match to return the library e.g. libraryname=</param>
-        /// <returns>instance of JavaScriptLibrary</returns>
+        /// <summary>Get information about the latest version of a <see cref="JavaScriptLibrary"/> that matches the given <paramref name="predicate"/></summary>
+        /// <param name="predicate">A function used to filter the library</param>
+        /// <example>
+        /// JavaScriptLibraryController.Instance.GetLibrary(l => string.Equals(l.LibraryName, "Knockout", StringComparison.OrdinalIgnoreCase))
+        /// </example>
+        /// <returns>The highest version <see cref="JavaScriptLibrary"/> instance that matches the <paramref name="predicate"/>, or <c>null</c> if no library matches</returns>
         public JavaScriptLibrary GetLibrary(Func<JavaScriptLibrary, bool> predicate)
         {
             return GetLibraries(predicate).OrderByDescending(l => l.Version).FirstOrDefault();
         }
 
-        /// <summary>
-        /// return collection of library references
-        /// </summary>
-        /// <param name="predicate">predicate to match to return the library collection e.g. libraryname=</param>
-        /// <returns>collection of library references</returns>
+        /// <summary>Gets all of the <see cref="JavaScriptLibrary"/> instances matching the given <paramref name="predicate"/></summary>
+        /// <param name="predicate">A function used to filter the library</param>
+        /// <example>
+        /// JavaScriptLibraryController.Instance.GetLibraries(l => string.Equals(l.LibraryName, "Knockout", StringComparison.OrdinalIgnoreCase))
+        /// </example>
+        /// <returns>A sequence of <see cref="JavaScriptLibrary"/> instances</returns>
         public IEnumerable<JavaScriptLibrary> GetLibraries(Func<JavaScriptLibrary, bool> predicate)
         {
             return GetLibraries().Where(predicate);
         }
 
-        /// <summary>
-        /// return all registered libraries
-        /// </summary>
-        /// <returns>collection of library references</returns>
+        /// <summary>Gets all of the <see cref="JavaScriptLibrary"/> instances</summary>
+        /// <returns>A sequence of <see cref="JavaScriptLibrary"/> instances</returns>
         public IEnumerable<JavaScriptLibrary> GetLibraries()
         {
 	    return CBO.GetCachedObject<IEnumerable<JavaScriptLibrary>>(new CacheItemArgs(DataCache.JavaScriptLibrariesCacheKey,
@@ -86,10 +84,8 @@ namespace DotNetNuke.Framework.JavaScriptLibraries
                                  c => CBO.FillCollection<JavaScriptLibrary>(DataProvider.Instance().ExecuteReader("GetJavaScriptLibraries")));
         }
 
-        /// <summary>
-        /// save a library to the database
-        /// </summary>
-        /// <param name="library">library to be saved</param>
+        /// <summary>Save a library to the database</summary>
+        /// <param name="library">Library to be saved</param>
         public void SaveLibrary(JavaScriptLibrary library)
         {
             library.JavaScriptLibraryID = DataProvider.Instance().ExecuteScalar<int>("SaveJavaScriptLibrary", 

